syncer/syncer.go

```diff
@@ -1741,7 +1759,11 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 			needHandleDDLs = appliedSQLs // maybe nil
 		}
 
-		job := newDDLJob(nil, needHandleDDLs, *ec.lastPos, *ec.currentPos, nil, nil, *ec.traceID)
+		if err := s.flushJobs(); err != nil {
```

Comment:
Do we need to `flushJobs()` before `TrySync` as we did in https://github.com/pingcap/dm/pull/625?